pennylane/transforms/qcut.py

```diff
@@ -675,3 +676,163 @@ def contract_tensors(
     kwargs = {} if use_opt_einsum else {"like": tensors[0]}
 
     return contract(eqn, *tensors, **kwargs)
+
+
+CHANGE_OF_BASIS = qml.math.array(
+    [[1.0, 1.0, 0.0, 0.0], [-1.0, -1.0, 2.0, 0.0], [-1.0, -1.0, 0.0, 2.0], [1.0, -1.0, 0.0, 0.0]]
+)
+
+
+def _process_tensor(results, n_prep: int, n_meas: int):
+    """Convert a flat slice of an individual circuit fragment's execution results into a tensor.
+
+    This function performs the following steps:
+
+    1. Reshapes ``results`` into the intermediate shape ``(4,) * n_prep + (4**n_meas,)``
+    2. Shuffles the final axis to follow the standard product over measurement settings. E.g., for
+      ``n_meas = 2`` the standard product is: II, IX, IY, IZ, XI, ..., ZY, ZZ while the input order
+      will be the result of ``qml.grouping.partition_pauli_group(2)``, i.e., II, IZ, ZI, ZZ, ...,
+      YY.
+    3. Reshapes into the final target shape ``(4,) * (n_prep + n_meas)``
+    4. Performs a change of basis for the preparation indices (the first ``n_prep`` indices) from
+       the |0>, |1>, |+>, |+i> basis to the I, X, Y, Z basis using ``CHANGE_OF_BASIS``.
+
+    Args:
+        results (array-like): the input execution results
```

Comment:
Quick flyby comment: in the rest of PL we use `tensor_like` to represent the set 
```
{np.ndarray, ArrayBox, qml.numpy.Tensor, tf.Tensor, tf.Variable, torch.Tensor, jax.DeviceArray}
```

Reply:
This is meant to mirror the `array_like` typehint that is used in NumPy, c.f., https://numpy.org/doc/stable/reference/generated/numpy.mean.html